Automattic/wp-calypso
# add experiment to add new site button

#### Changes proposed in this Pull Request

Add experiment to just the add-site button, see pbxNRc-1jD-p2

fixes #60166 

The new experiments platform code uses [react hooks to get experiments](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/explat/README.md), since we already have a somewhat involved function for getting the onboardingUrl, I created a combined useOnboardingUrl hook which could take all of the neccessary logic into account.

There are other links to create a new site, but I don't believe these need to be updated (I got a bit lost in the weeds trying to do that) I don't think it's necessary to update all links. I had a think about whether it will be neccessary to add new events to use as assignment filters. pbxNRc-1jD-p2#comment-3011

#### Testing instructions

Open the "Switch Site" menu, the "Add New Site" button at the bottom should take you to `/new` following existing logic. If 
Related to #60166
